xcdat/axis.py

```diff
@@ -0,0 +1,245 @@
+from typing import List, Literal, TypedDict
+
+import numpy as np
+import xarray as xr
+
+from xcdat.xcdat import logger
+
+
+@xr.register_dataset_accessor("axis")
+class AxisAccessor:
+    """A class used to represent an AxisAccessor (xarray dataset accessor)
+
+    :param xarray_obj: The Dataset object to be extended
+    :type xarray_obj: xr.Dataset
+
+    Examples
+    --------
+    Get bounds:
+
+    >>> from xcdat import axis
+    >>> ds = xr.open_dataset("file_path")
+    >>> # Get generated bounds
+    >>> lat_bnds = ds.axis.get_bounds('lat', generate=True)
+    >>> lon_bnds = ds.axis.get_bounds('lon', generate=True)
+    >>> # Get existing bounds
+    >>> lat_bnds = ds.axis.get_bounds('lat')
+    >>> lon_bnds = ds.axis.get_bounds('lon')
+    """
+
+    # Type annotation definitions
+    Axis = Literal["lat", "lon"]
+    AxesMapValue = TypedDict(
+        "AxesMapValue", {"coords": List[str], "bounds_vars": List[str]}
+    )
+    AxesMap = TypedDict("AxesMap", {"lat": AxesMapValue, "lon": AxesMapValue})
+
+    # Mapping of generic axes names to coordinates and bounds variables in the Dataset
+    axes_map: AxesMap = {
+        "lat": {
+            "coords": ["lat", "latitude"],
+            "bounds_vars": ["lat_bnds", "latitude_bnds"],
+        },
+        "lon": {
+            "coords": ["lon", "longitude"],
+            "bounds_vars": ["lon_bnds", "longitude_bnds"],
+        },
+    }
+
+    def __init__(self, xarray_obj: xr.Dataset):
+        self._dataset = xarray_obj
+
+    def get_bounds(self, axis: Axis, generate: bool = False) -> xr.DataArray:
+        """Get bounds for an axis.
+
+        It will either generate the bounds or return the existing bounds.
+
+        :param axis: "lat" or "lon" axis
+        :type axis: Axis
+        :param generate: If True, generate the bounds regardless if it exists in the
+        dataset or not. If False and the bounds already exist, return the existing
+        bounds, defaults to False
+        :type generate: bool, optional
+        :return: Axis bounds
+        :rtype: xr.DataArray
+        """
+        if generate:
+            logger.info(f"Generating bounds for {axis} axis")
+            return self._gen_bounds(axis)
+
+        bounds_vars = AxisAccessor.axes_map[axis]["bounds_vars"]
+        matching_bounds_var = None
+
+        for var in bounds_vars:
+            matching_bounds_var = self._dataset.data_vars.get(var)
+            if matching_bounds_var is not None:
+                logger.info(f"Returning existing {axis} bounds from the dataset.")
+                return matching_bounds_var
+
+        raise ValueError(
+            f"{axis} bounds not found in the dataset. Pass generate=True to generate bounds."
+        )
```

Comment:
If bounds don't exist and they user doesn't pass `generate=True`, it will throw a ValueError.

Reply:
Removing this to avoid frustrating runtime errors for most use cases.